Mathlib/NumberTheory/LSeries/Coercion.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.ArithmeticFunction
+import Mathlib.Analysis.Complex.Basic
+
+/-!
+# Coercion to complex-valued arithmetic functions
+
+This file defines coercions from arithmetic functions with values in some commutative
+(semi)ring `R` to arithmetic functions with values in `ℂ`, given that `ℂ` is an `R`-algebra.
+This is necessary to be able to write `LSeries f` when `f : ArithmeticFunction ℤ` (say),
+as `LSeries` requires an arithmetic function with values in `ℂ` as input.
+-/
+
+namespace ArithmeticFunction
+
+section toComplex
+
+/-!
+### Definition and compatibility with operations
+-/
+
+variable {R : Type*} [CommSemiring R] [Algebra R ℂ]
+
+/-- Coerce an arithmetic function with values in `R` to one with values in `ℂ`.
+We cannot inline this in `instToComplexAritmeticFunction` because it gets unfolded too much. -/
+@[coe] def toComplexArithmeticFunction (f : ArithmeticFunction R) : ArithmeticFunction ℂ where
+  toFun n := algebraMap R ℂ (f n)
+  map_zero' := by simp only [map_zero, _root_.map_zero]
+
+instance instToComplexAritmeticFunction : CoeHead (ArithmeticFunction R) (ArithmeticFunction ℂ) :=
```

Comment:
Changed the name.

After reading the [docs on Coe](https://leanprover-community.github.io/mathlib4_docs/Init/Coe.html#Important-typeclasses) again, I thinkt that `CoeHead` should be the right one (`{R}` is a superset of the empty set, and since `ℂ` is an algebra over itself, the instance could potentially loop).